spec/factories/annual_leave_request.rb

```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory(:annual_leave_request) do
+    date_from { Time.zone.today + 7 }
+    date_to { Time.zone.today + 14 }
+    days_required { WorkingDaysCalculator.new(date_from, date_to).number_of_working_days }
+    user_id { create(:user).id }
```

Comment:
A general tip with factories, and people might disagree with this but I'd stand by it: I wouldn't create associated models inside factory definitions. It feels like extra work to create them in your tests instead, but over time you'll appreciate the flexibility of being able to create a user with whatever attributes you want to use, the readability of being able to see your whole test fixture setup in your tests and also the performance benefit of not making any database calls you don't need to.